.github/workflows/ci.yml

```diff
@@ -0,0 +1,156 @@
+# CI WorkFlow for build, test and release of Plugin Packages
+name: CI for Plugin Packages
+on:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - 'v*.*.*'
+    paths-ignore:
+      - 'plugins/**'
+      - 'hack/test-plugin-manifests.sh'
+      - 'hack/update-plugin-manifests.sh'
+      - 'hack/update-preflight-manifest.sh'
+      - 'hack/update-log-collector-manifest.sh'
+      - '.github/workflows/plugins.yml'
+  pull_request:
+    branches:
+      - '*'
+    paths-ignore:
+      - 'plugins/**'
+      - 'hack/test-plugin-manifests.sh'
+      - 'hack/update-plugin-manifests.sh'
+      - 'hack/update-preflight-manifest.sh'
+      - 'hack/update-log-collector-manifest.sh'
+      - '.github/workflows/plugins.yml'
+
+jobs:
+  pre-build:
+    runs-on: ubuntu-latest
+    outputs:
+      preflight: ${{ steps.filter.outputs.preflight }}
+      logCollector: ${{ steps.filter.outputs.logCollector }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Run yamllint
+        uses: reviewdog/action-yamllint@v1
+        with:
+          fail_on_error: true
+          reporter: local
+          filter_mode: nofilter
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.29
+
+      - name: Run Shell lint
+        run: hack/run-shell-lint.sh
+
+      - name: Verify code patterns
```

Comment:
check if we can remove this and do these things via golangci-lint

Reply:
sure.. will do that